driver_4_lights.ino

```diff
@@ -1,225 +1,345 @@
+/*
+  State button (LED to Human)       meaning
+  0 Nothing (LED 1/sec blink)       Just looping looking for input
+  1 StepLights (LED 3/sec blink)    Can step through lights
+  2 ManualStepThrough (LED 10/sec)  When you push camera button, light on and picture
+  3 AutoStepThrough (LED on)        Goes through all lights and all pictures
 
-  modified 8 May 2014
-  by Scott Fitzgerald
+  If you are in the middle of a state, pressing state returns you to state 0 (Nothing)
 
-  modified 2 Sep 2016
-  by Arturo Guadalupi
+  Camera button: Push to take picture while in state
+  1 (StepLights) changes the lights
+  2 (ManualStepThrough) change the lights and camera
+  3 (AutoStepThrough) collect pictures with light automatically for all lights
 
-  modified 8 Sep 2016
-  by Colby Newman
 */
 
-
-// constants won't change. They're used here to
-// set pin numbers:
-const int pinButtonState = 0;    // the number of the pushbutton pin
-const int pinButtonCamera = 8;   //button for the camera
-const int ledPin = 13;      // the number of the LED pin
+const int pinButtonState = 8; // the number of the pushbutton pin
+const int pinButtonCamera = 10; //button for the camera
+const int ledPin = 13; // the number of the LED pin
+const int indicatorLed = 12; //the led that shows what state you are in on circuit board
+const int cameraShutter = 11;
+const int ledTR = 3;   //TR light on circut board
+const int ledTL = 4;   //TL light on cirut board
+const int ledBR = 5;   //BR light on circut board
+const int ledBL = 6;   //BL light on circut board
+const int ledAll = 10; // all the lights are ON (not on circut board)
+
+const int whichLightOn; //which of the 4 lights are on
+
+const int shutterLag = 50;
+const int shutterDelay = 100;
 const int ON = 1;
 const int OFF = 0;
-const int all = 10;           // all lights ON
-const int topright = 20;      // top right is ON
-const int topleft = 30;
-const int bottomleft = 40;
-const int bottomright = 50;
+
+//for cameraButton
+const int stepLights = 1; // changes the lights with no pictures taken
+const int manualStepThrough = 2; // change the lights and camera
+const int autoStepThrough = 3; // collect pictures with light automatically for all lights in sequence
 
 // Variables will change:
-int ledState = HIGH;         // the current state of the output pin
-int buttonPinState;             // the current reading from the input pin
-int buttonCameraState;
-int lastButtonPinState = LOW;   // the previous reading from the input pin
-int lastButtonCameraState = LOW;
-int lightState;
-int clickState;
+int ledState = HIGH; // the current state of the output pin
+int buttonState = HIGH; // the current reading from the input pin
+int buttonCamera = HIGH; //so it does not start w/out button click
+int lastButtonState = HIGH; // the previous reading from the input pin
+int lastButtonCamera = HIGH;
+int state = 0; //0 Do nothing, 1 StepLIghts, 2 ManualStepThrough, 3 AutoStepThrough
```

Comment:
To make your code more readable use an enum.
http://en.cppreference.com/w/cpp/language/enum

e.g.
enum class LightState { WAIT, STEP_LIGHTS, MANUAL_STEP_THROUGH, AUTO_STEP_THROUGH };